pallets/registrar/src/lib.rs

```diff
@@ -902,12 +988,22 @@ pub mod pallet {
                 Pallet::<T>::registered_para_ids()
             };
 
-            paras.into_iter().collect()
-        }
+            let mut parachains = vec![];
+            let mut parathreads = vec![];
 
-        fn session_parathreads(_session_index: T::SessionIndex) -> Vec<ParaId> {
-            // FIXME(parathreads)
-            vec![]
+            for para_id in paras {
+                // TODO: sweet O(n) db reads
+                if let Some(parathread_params) = ParathreadParams::<T>::get(&para_id) {
```

Comment:
Here we need some way to distinguish parathreads from parachains. I am using the `ParathreadParams` storage map, but that will do N reads. Here are some alternatives, not sure what's the best one because I don't yet know if params will be used on-chain or not. They will probably be updated over time, so that's why I initially choose the StorageMap. Alternatives:

```
* StorageMap<ParaId, Params> (current)
* StorageValue<BTreeMap<ParaId, Params>> (all under one key)
* StorageValue<Vec<ParaId>> and StorageMap<ParaId, Params> (combination of both)
* Refactor currentContainerChains from `Vec<ParaId>` to `Vec<(ParaId, IsParathread)>`
```

Reply:
Ideally we should not need another storage item for this. It is very likely that every time we read `RegisteredParchains` we will need to know whether they are a `Parachain` or a `Parathread`, so we might just want to store a struct indicating this already in `RegisteredParchains`...

Reply:
We decided to keep the current StorageMap, but adding a `PendingParams = StorageValue<Vec<(ParaId, Params)>>` so that changes are only reflected in the next session.

We also would like to change `RegisteredParchains` into a `StorageValue<Vec<(ParaId, Option<Params>)>>`, where `Params` includes both `Option<ParathreadParams>` and also other things such as `paused: bool`. But that looks like a big change so it will be done in a future PR.